### ISSUE TITLE
Symbolic link /etc/localtime not managed correctly

### ISSUE THREAD

**Issue by smoeding** [saz/puppet-timezone#85]
The creaton of the symbolic link `/etc/localtime` doesn't work correctly in the latest release.

I have a server where `/etc/localtime` is a regular file and the module does not create a symbolic link to replace the file. The cause is probably the file resource that manages the link:

```
file { $localtime_file:
  ensure => $localtime_ensure,
  source => "${zoneinfo_dir}/${timezone}",
  notify => $notify_services,
}
```

The documentation of the file resource says:

> link ensures the file is a symlink, and requires that you also set the target attribute

So `source` should probably be `target` here.

**Comment by saz** [comment]
Oh man... This is more complicated.
Looking at https://wiki.debian.org/TimeZoneChanges it should be a file and not a link.
Same for e.g. FreeBSD or Gentoo.

And then we have...
```
root@northstar:~# ls -la /etc/localtime 
lrwxrwxrwx 1 root root 33 Jan  3 06:25 /etc/localtime -> /usr/share/zoneinfo/Europe/Berlin
root@northstar:~# rm /etc/localtime 
root@northstar:~# timedatectl set-timezone Europe/Berlin
root@northstar:~# ls -la /etc/localtime 
lrwxrwxrwx 1 root root 35 Feb 15 14:58 /etc/localtime -> ../usr/share/zoneinfo/Europe/Berlin
root@northstar:~#
```

**Comment by saz** [comment]
I'll stick with the symlink. It fits most systems.

Should be fixed in v5.1.1
